src/effect.ts

```diff
@@ -34,15 +34,19 @@ export class Effect {
   }
 
   compute(): (() => void) | void {
-    const prevCompute = MANAGER.currentCompute;
-    MANAGER.currentCompute = new Set();
+    const prevCompute = MANAGER.currentContext;
+
+    const context = MANAGER.fetchContext(this);
+    context.clear();
+    MANAGER.currentContext = context;
+
     MANAGER.runningEffect = this;
 
     this.#computeDeps();
 
     if (this.#prevTags && getMax(this.#prevTags) === this.#version) {
       MANAGER.runningEffect = null;
-      MANAGER.currentCompute = prevCompute;
+      MANAGER.currentContext = prevCompute;
```

Comment:
Thinking about this API, and the similar one in `Derived`, continues to make me skittish about having these manage the `Manager` state themselves. Definitely a "revisit later" thing, but having the manager, err, manage, seems good?

Reply:
It's likely that `Manager` is just a bad name. It originally started as `Version` because it was just a singleton container for the version itself, but as it grew in use I renamed it to `Manager` which seemed just slightly better :joy:

There's likely some improvement we could do here, but whenever I start to move too much of the actual state manipulation into the manager itself, it feels like it starts to turn into E_TOO_MUCH_INDIRECTION very quickly

Reply:
> It's likely that Manager is just a bad name.

That's interesting; I'll mull on both a better name and whether/how we could avoid the `E_TOO_MUCH_INDIRECTION` problem—I had a similar thought about that problem when thinking about this, but it *also* feels a little bad to have all these other classes directly manipulating the singleton's state. Might have to take a few swings at it to see if there's something which improves on *that* part but avoids the indirection. 🤔